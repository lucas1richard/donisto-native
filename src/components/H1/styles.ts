import { StyleSheet, TextStyle } from 'react-native';
import theme from 'theme/iftheme';

const defaultStyles: TextStyle = {
  fontSize: theme.fontSize * 2,
  color: theme.black,
  textAlign: 'center'
};

export default StyleSheet.create({
  default: {
    ...defaultStyles
  },
  primary: {
    ...defaultStyles,
    color: theme.primary[400]
  },
  secondary: {
    ...defaultStyles,
    color: theme.secondary[400]
  },
  light: {
    ...defaultStyles,
    color: '#fff'
  }
});
