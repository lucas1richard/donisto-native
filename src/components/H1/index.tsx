import * as React from 'react';
import { Text, TextProperties } from 'react-native';
import styles from './styles';

export interface IH1Props extends TextProperties {
  children: string;
  color?: 'primary'|'secondary'|'default'|'light';
}

const H1 = ({ children, color, style }: IH1Props) => {
  return (
    <Text style={[styles[color], style]}>
      {children}
    </Text>
  );
};

H1.defaultProps = {
  color: 'default'
};

export default H1;
