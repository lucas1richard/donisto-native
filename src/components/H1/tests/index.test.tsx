import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import H1, { IH1Props } from '..';

describe('<H1 />', () => {
  const renderComponent = (props: IH1Props) => toJson(shallow(<H1 {...props} />));
  it('renders with just text', () => {
    expect(renderComponent({ children: 'mockText' })).toMatchSnapshot();
  });
  it('renders with a color prop', () => {
    expect(renderComponent({
      children: 'mockText',
      color: 'light',
    })).toMatchSnapshot();
  });
});