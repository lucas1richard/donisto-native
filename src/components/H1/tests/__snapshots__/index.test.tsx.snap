// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> renders with a color prop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <H1
    color="light"
  >
    mockText
  </H1>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessible": true,
      "allowFontScaling": true,
      "children": "mockText",
      "ellipsizeMode": "tail",
      "style": Array [
        Object {
          "color": "#fff",
          "fontSize": 34,
          "textAlign": "center",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": "mockText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "mockText",
        "ellipsizeMode": "tail",
        "style": Array [
          Object {
            "color": "#fff",
            "fontSize": 34,
            "textAlign": "center",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": "mockText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<H1 /> renders with just text 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <H1
    color="default"
  >
    mockText
  </H1>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessible": true,
      "allowFontScaling": true,
      "children": "mockText",
      "ellipsizeMode": "tail",
      "style": Array [
        Object {
          "color": "#000000",
          "fontSize": 34,
          "textAlign": "center",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": "mockText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "mockText",
        "ellipsizeMode": "tail",
        "style": Array [
          Object {
            "color": "#000000",
            "fontSize": 34,
            "textAlign": "center",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": "mockText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
